### PR TITLE
TRAIN-214: Create Model Relationships

### PR DESCRIPTION
**Pull request by r-barlow**

### CODE REVIEW

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Account.java [hunk 1]
```diff
@@ -16,6 +20,9 @@ public class Account {
     @Column
     private String name;
 
+    @OneToMany(mappedBy = "account")
+    private List<Category> categories =  new ArrayList<>();
```
Comment (colsen-neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Account.java]: Rather don't assign a default value here, you can rather perform a null check in the accessor and assign a new array list to this field if it is null.

Comment (colsen-neslo) [reply src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Account.java]: Please make this change across all collections in all models where you have assigned a default value.

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Account.java [hunk 2]
```diff
@@ -31,4 +38,12 @@ public String getName() {
     public void setName(final String name) {
         this.name = name;
     }
+
+    public List<Category> getCategories() {
+        return categories;
+    }
```
Comment (colsen-neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Account.java]:
```suggestion
    public List<Category> getCategories() {
        if (categories == null) {
            return new ArrayList<>();
        }
        
        return categories;
    }
```

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Transaction.java [hunk 3]
```diff
@@ -24,6 +25,9 @@ public class Transaction {
     @Column(length = 19, precision = 4)
     private Double amount;
 
+    @ManyToOne
+    private Transaction transaction;
```
Comment (colsen-neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/model/Transaction.java]: Is this correct to say many transactions can have a single transaction? I don't quite understand this relationship?

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

**Review by JOlsen-Neslo — APPROVED** [review]

**Review by colsen-neslo — APPROVED** [review]

**Pull request merged**